models/user.go

```diff
@@ -237,7 +237,7 @@ func (u *User) CanCreateOrganization() bool {
 
 // CanEditGitHook returns true if user can edit Git hooks.
 func (u *User) CanEditGitHook() bool {
-	return u.IsAdmin || u.AllowGitHook
+	return !setting.DisableGitHooks && (u.IsAdmin || u.AllowGitHook)
```

Comment:
Please add this check to the API to if needed. 
https://github.com/go-gitea/gitea/blob/04ec79579cdcd5cf647ef3ff33d35ecfa7550b49/routers/api/v1/api.go#L377

Reply:
Thanks for this feedback, especially linking me to where I could look. Investigating this (by calling the API to see what the response would be), I found that it only returns the standard gitea/slack/discord/etc.. hooks, and couldn't find it returning the githooks.